### DISCUSSION TITLE
E-series CIC

### DISCUSSION THREAD

**Discussion by 85michal** [BimmerGestalt/AAIdrive#241] [General]
First thing I must say, you did amazing job with this project and it is highly appreciated. I run E-series CIC infotainment, without the Combox needed for Connected Drive yet. I was wondering if anyone contributing here ever tried to see if this project works with CIC. As far as my research is valid, there were no 3rd party apps integrated wtih CIC via android devices, however there were integrations based on iOS, such as Spotify. Digging the web I found some info there was an SDK dedicated for 3rd parties released by the BMW already at that time: https://youtu.be/wzGnzWgWTAg

I just wonder if you think there might be any chance of using/adapting this project for CIC.

Happy Easter to all of You.

**Comment by hufman** [comment]
Thank you for that video, it was really neat!
I've tried some USB sniffing experiments with an IPhone and an older BMW, and it seems that the protocol is the same (an Etch connection per app, multiplexed in a single BCL stream), but wrapped within the iPhone iAP accessory protocol instead of the Android accessory protocol. I don't think Android itself would be able to talk this USB protocol without a great deal of license-violating kernel work (or else, surely an Android phone or app would have come out with support for all the existing IPhone accessories). I think some sort of little Arduino-style device would need to sit between the car and the phone and translate between the two USB protocols.
There might need to be a tweak in the app's authentication step too, I know there's some difference that is preventing using an IPhone app's cert in an Android app.
But I think the rest of the protocol is the same!
